config.go

```diff
@@ -7,9 +7,74 @@ import (
 	"time"
 )
 
+// These are the versions of the protocol (which includes RPC messages as
+// well as Raft-specific log entries) that this server can _understand_. Use
+// the ProtocolVersion member of the Config object to control the version of
+// the protocol to use when _speaking_ to other servers. This is not currently
+// written into snapshots so they are unversioned. Note that depending on the
+// protocol version being spoken, some otherwise understood RPC messages may be
+// refused. See isVersionCompatible for details of this logic.
+//
+// There are notes about the upgrade path in the description of the versions
+// below. If you are starting a fresh cluster then there's no reason not to
+// jump right to the latest protocol version. If you need to interoperate with
+// older, unversioned Raft servers you'll need to drive the cluster through the
+// different versions in order. This may best be done by moving forward across a
+// series of releases of your application.
+//
+// Version History
+//
+// 0: Unversioned, original protocol, used to interoperate with unversioned
+//    Raft servers. Under this version all configuration changes are propagated
+//    using the deprecated RemovePeerDeprecated Raft log entry. This means that
+//    server IDs are always set to be the same as the server addresses (since
+//    the old log entry type cannot transmit an ID), and only AddPeer/RemovePeer
+//    APIs are supported. This version can understand the new LogConfiguration
+//    Raft log entry but it will never generate one.
```

Comment:
How could this version possibly understand the new LogConfiguration entries? They're not part of `master` today.